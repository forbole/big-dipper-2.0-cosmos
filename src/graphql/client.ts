import WebSocket from 'isomorphic-ws';
import {
  ApolloClient,
  InMemoryCache,
  split,
  HttpLink,
  ApolloLink,
  concat,
} from '@apollo/client';
import {
  getMainDefinition,
} from '@apollo/client/utilities';
import { WebSocketLink } from '@apollo/client/link/ws';

import { useMemo } from 'react';

const defaultOptions:any = {
  watchQuery: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
  query: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
};

let apolloClient;

const httpLink = new HttpLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_URL,
});

const wsLink = new WebSocketLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_WS ?? 'wss://localhost:3000',
  options: {
    reconnect: true,
  },
  webSocketImpl: WebSocket,
});

const link = typeof window !== 'undefined' ? split(
  ({ query }) => {
    const {
      kind, operation,
    }:any = getMainDefinition(query);
    return kind === 'OperationDefinition' && operation === 'subscription';
  },
  wsLink,
  httpLink,
) : httpLink;

const authMiddleware = new ApolloLink((operation, forward) => {
  operation.setContext({
    headers: {
    },
  });

  return forward(operation);
});

function createApolloClient() {
  const client = new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: concat(authMiddleware, link),
    cache: new InMemoryCache({
    }),
  });

  client.defaultOptions = defaultOptions;

  return client;
}

export function initializeApollo(initialState = null) {
  // eslint-disable-next-line
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();
    // Restore the cache using the data passed from getStaticProps/getServerSideProps
    // combined with the existing cached data
    _apolloClient.cache.restore({
      ...existingCache, ...initialState,
    });
  }
  // For SSG and SSR always create a new Apollo Client
  if (typeof window === 'undefined') return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState) {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
