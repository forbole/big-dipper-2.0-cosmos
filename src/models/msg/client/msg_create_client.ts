import { Categories } from '../types';

class MsgCreateClient {
    public category: Categories;
    public type: string;
    public signer: string;
    public chainId: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.chainId = payload.chainId;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgCreateClient({
        json,
        type: json['@type'],
        signer: json.signer,
        chainId: json.client_state?.chain_id,
      });
    }
}

export default MsgCreateClient;
