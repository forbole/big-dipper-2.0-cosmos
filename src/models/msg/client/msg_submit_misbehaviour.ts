import { Categories } from '../types';

class MsgSubmitMisbehaviour {
    public category: Categories;
    public type: string;
    public signer: string;
    public clientId: string;
    public misbehaviour: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.clientId = payload.clientId;
      this.misbehaviour = payload.misbehaviour;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgSubmitMisbehaviour({
        json,
        type: json['@type'],
        signer: json.signer,
        clientId: json.client_id,
        misbehaviour: json.misbehaviour,
      });
    }
}

export default MsgSubmitMisbehaviour;
