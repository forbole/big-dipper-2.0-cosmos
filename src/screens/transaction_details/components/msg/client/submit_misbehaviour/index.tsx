import React from 'react';
import Trans from 'next-translate/Trans';
import { Typography } from '@material-ui/core';
import { Name } from '@components';
import { MsgSubmitMisbehaviour } from '@models';
import { useChainContext } from '@contexts';

const SubmitMisbehaviour = (props: {
    message: MsgSubmitMisbehaviour;
}) => {
  const { findAddress } = useChainContext();
  const { message } = props;

  const signer = findAddress(message.signer);
  const signerMoniker = signer ? signer?.moniker : message.signer;

  return (
    <Typography>
      <Trans
        i18nKey="message_contents:txSubmitMisbehaviourContent"
        components={[
          (
            <Name
              address={message.signer}
              name={signerMoniker}
            />
          ),
          <b />,
        ]}
        values={{
          clientId: message.clientId,
        }}
      />
    </Typography>
  );
};

export default SubmitMisbehaviour;
