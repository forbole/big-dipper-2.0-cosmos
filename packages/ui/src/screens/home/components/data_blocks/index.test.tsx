import { ApolloClient, ApolloProvider, from, InMemoryCache } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';
import {
  ActiveValidatorCountDocument,
  AverageBlockTimeDocument,
  LatestBlockHeightListenerDocument,
  TokenPriceListenerDocument,
} from '@/graphql/types/general_types';
import DataBlocks from '@/screens/home/components/data_blocks';
import MockTheme from '@/tests/mocks/MockTheme';
import wait from '@/tests/utils/wait';

// ==================================
// mocks
// ==================================
beforeEach(() => {
  jest.mock(
    '@/screens/home/components/data_blocks/components/single_block',
    () => (props: JSX.IntrinsicElements['div']) => <div id="SingleBlock" {...props} />
  );
});

const mockLatestBlockHeight = jest.fn().mockReturnValue({
  data: {
    height: [
      {
        height: 953992,
      },
    ],
  },
});

const mockAverageBlockTime = jest.fn().mockReturnValue({
  data: {
    averageBlockTime: [
      {
        averageTime: 6.540624503709312,
      },
    ],
  },
});

const mockTokenPrice = jest.fn().mockReturnValue({
  data: {
    tokenPrice: [],
  },
});

const mockActiveValidatorsCount = jest.fn().mockReturnValue({
  data: {
    activeTotal: {
      aggregate: {
        count: 200,
      },
    },
    inactiveTotal: {
      aggregate: {
        count: 332,
      },
    },
    total: {
      aggregate: {
        count: 532,
      },
    },
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Home/DataBlocks', () => {
  it('matches snapshot', async () => {
    let component: renderer.ReactTestRenderer | undefined;

    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={new ApolloClient({ link: from([]), cache: new InMemoryCache() })}>
          <MockedProvider
            mocks={[
              {
                request: { query: LatestBlockHeightListenerDocument, variables: { offset: 0 } },
                result: mockLatestBlockHeight,
              },
              { request: { query: TokenPriceListenerDocument }, result: mockTokenPrice },
              { request: { query: AverageBlockTimeDocument }, result: mockAverageBlockTime },
              {
                request: { query: ActiveValidatorCountDocument },
                result: mockActiveValidatorsCount,
              },
            ]}
          >
            <MockTheme>
              <DataBlocks />
            </MockTheme>
          </MockedProvider>
        </ApolloProvider>
      );
    });
    await wait(renderer.act);
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockActiveValidatorsCount).toBeCalledTimes(1);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
