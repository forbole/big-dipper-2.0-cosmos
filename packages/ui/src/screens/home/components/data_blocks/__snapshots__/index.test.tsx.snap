// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/DataBlocks matches snapshot 1`] = `
.emotion-0 {
  display: grid;
  gap: 8px;
  grid-template-rows: auto;
}

@media (min-width:375px) {
  .emotion-0 {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media (min-width:1280px) {
  .emotion-0 {
    gap: 16px;
    grid-template-columns: repeat(4, 1fr);
  }
}

.emotion-1 {
  padding: 16px;
  background: #FF835B;
  border-radius: 4px;
  height: 110px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  color: #FFFFFF;
  background: #F87255;
}

.emotion-1 .label {
  margin-bottom: 16px;
}

.emotion-1 .content {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-1 .description {
  display: none;
}

@media (min-width:768px) {
  .emotion-1 .description {
    display: block;
  }
}

.emotion-1 .flexContent {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 4px;
}

.emotion-2 {
  margin: 0;
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.43;
}

.emotion-3 {
  margin: 0;
  font-size: 2rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  line-height: 1.167;
}

.emotion-4 {
  padding: 16px;
  background: #FF835B;
  border-radius: 4px;
  height: 110px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  color: #FFFFFF;
  background: #FA9147;
}

.emotion-4 .label {
  margin-bottom: 16px;
}

.emotion-4 .content {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-4 .description {
  display: none;
}

@media (min-width:768px) {
  .emotion-4 .description {
    display: block;
  }
}

.emotion-4 .flexContent {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 4px;
}

.emotion-7 {
  padding: 16px;
  background: #FF835B;
  border-radius: 4px;
  height: 110px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  color: #FFFFFF;
  background: #20D494;
}

.emotion-7 .label {
  margin-bottom: 16px;
}

.emotion-7 .content {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-7 .description {
  display: none;
}

@media (min-width:768px) {
  .emotion-7 .description {
    display: block;
  }
}

.emotion-7 .flexContent {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 4px;
}

.emotion-10 {
  margin: 0;
  font-size: 0.75rem;
  letter-spacing: 0.4px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.66;
}

.emotion-11 {
  padding: 16px;
  background: #FF835B;
  border-radius: 4px;
  height: 110px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  color: #FFFFFF;
  background: #2FB6E0;
}

.emotion-11 .label {
  margin-bottom: 16px;
}

.emotion-11 .content {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-11 .description {
  display: none;
}

@media (min-width:768px) {
  .emotion-11 .description {
    display: block;
  }
}

.emotion-11 .flexContent {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 4px;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 label emotion-2"
    >
      latestBlock
    </p>
    <div
      className="content"
    >
      <h1
        className="MuiTypography-root MuiTypography-h1 emotion-3"
      >
        953,992
      </h1>
      <div
        className="flexContent"
      />
    </div>
  </div>
  <div
    className="emotion-4"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 label emotion-2"
    >
      averageBlockTime
    </p>
    <div
      className="content"
    >
      <h1
        className="MuiTypography-root MuiTypography-h1 emotion-3"
      >
        6.54 s
      </h1>
      <div
        className="flexContent"
      />
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 label emotion-2"
    >
      price
    </p>
    <div
      className="content"
    >
      <h1
        className="MuiTypography-root MuiTypography-h1 emotion-3"
      >
        N/A
      </h1>
      <div
        className="flexContent"
      >
        <span
          className="MuiTypography-root MuiTypography-caption description emotion-10"
        >
          dataFrom
        </span>
        <svg />
      </div>
    </div>
  </div>
  <div
    className="emotion-11"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 label emotion-2"
    >
      activeValidators
    </p>
    <div
      className="content"
    >
      <h1
        className="MuiTypography-root MuiTypography-h1 emotion-3"
      >
        200
      </h1>
      <div
        className="flexContent"
      >
        <span
          className="MuiTypography-root MuiTypography-caption description emotion-10"
        >
          outOfValidators
        </span>
      </div>
    </div>
  </div>
</div>
`;
